shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -176,24 +181,34 @@ private void processingComplete(Class<?> entity) {
      * @throws IllegalStateException in case of an invalid HTTP response
      */
     private void validateHttpResponseBeforeParsing(HttpResponse<Buffer> httpResponse) {
+        if (httpResponse.statusCode() == 401) {
+            LOGGER.debug("Request to the manager was unauthorized.");
+            threadContext.contextualRunnable(this::refreshTokens);
```

Comment:
what is the purpose of this?

Reply:
@kostola you mean `threadContext.contextualRunnable` or the refresh of the token when we get unauthorized?

Reply:
the use of `threadContext.contextualRunnable`

Reply:
Because `validateHttpResponseBeforeParsing` is running on a thread that can not block